.jsonschema.json

```diff
@@ -0,0 +1,156 @@
+{
+  "title": "Simple Icons",
+  "definitions": {
+    "brand": {
+      "description": "A single brand",
+      "type": "object",
+      "required": ["title", "hex", "source"],
+      "properties": {
+        "title": {
+          "description": "The name of the brand",
+          "type": "string"
+        },
+        "hex": {
+          "description": "The brand color as a 6-character value (without #)",
+          "type": "string",
+          "pattern": "^[0-9A-F]{6}$"
+        },
+        "source": {
+          "description": "The website from which the icon was sourced",
+          "$ref": "#/definitions/url"
+        },
+        "guidelines": {
+          "description": "The brand guidelines",
+          "$ref": "#/definitions/url"
+        },
+        "license": {
+          "description": "The license for the icon",
+          "oneOf": [
+            {
+              "type": "object",
+              "required": ["type", "url"],
+              "properties": {
+                "type": {
+                  "description": "The license name or 'custom'",
+                  "type": "string",
+                  "enum": ["custom"]
+                },
+                "url": {
+                  "description": "The URL to the license text by the brand",
+                  "$ref": "#/definitions/url"
+                }
+              },
+              "additionalProperties": false
+            },
+            {
+              "type": "object",
+              "required": ["type"],
+              "properties": {
+                "type": {
+                  "description": "An SPDX License Identifier",
+                  "type": "string",
+                  "enum": [
```

Comment:
Since this implementation already uses a script to run the linting process, why not just download [this JSON file](https://github.com/spdx/license-list-data/blob/master/json/licenses.json) (extracted from [SPDX web repository](https://github.com/spdx/license-list-data)) and build the enum in-place?

Reply:
I'm not totally against it but in my opinion this is still not a good solution. First, the fact that we have a script now does not mean that we should stick with it, and I'm not sure if being able to pull this list dynamically is a good reason. Secondly, at least the repository you linked to, does not interface well with NPM. Sure, we _could_ pull the git repo manually or even the file directly, but we have no guarantees on the structure of the repository, hence this (the JSON linter!) might break overnight just because they changed something (*).

What we could do instead is have some automation that pulls the list on a schedule and updates the list for is. Still not ideal, but at least the `npm run jsonlint` script never breaks - if the scheduled workflow breaks that doesn't affect anyone directly.


---


(*): I'm not so worried that the structure of the JSON file itself changes, but this is a concern as well and ideally we have a source that provides a JSON schema for the JSON data they're providing us. Speaking of which, we may consider providing our JSON schema alongside `_data/simple-icons.json` in the NPM package 🤔

Reply:
Can't believe I didn't think of this earlier, but we should be pretty much fine with an approach like this if we use [this NPM package](https://www.npmjs.com/package/spdx-license-ids).

> First, the fact that we have a script now does not mean that we should stick with it, and I'm not sure if being able to pull this list dynamically is a good enough reason to stick with it.

Coming back to this, I guess I changed my mind and if it benefits us to have a custom CLI interface, than using a script for that is worthwhile, I think... Though it would be even more ideal if a potential CLI for [jsonschema](https://www.npmjs.com/package/jsonschema) supported `.js` config files so that the configuration can be constructed dynamically.

In that regard, do we want to change [`.jsonschema.json`](https://github.com/simple-icons/simple-icons/blob/4c9cb1102f36c7f0af544e6eb19686d680d5f39c/.jsonschema.json) to `.jsonschema.js` and handle the logic there with [the NPM package](https://www.npmjs.com/package/spdx-license-ids) there, or inject it's data in [the `jsonlint.js` script](https://github.com/simple-icons/simple-icons/blob/4c9cb1102f36c7f0af544e6eb19686d680d5f39c/scripts/jsonlint.js)?

Reply:
@PeterShaggyNoble @mondeja